receiver.go

```diff
@@ -441,15 +451,18 @@ func newReceiver(source string, session *Session, opts *ReceiverOptions) (*Recei
 		return r, nil
 	}
 
-	r.batching = opts.Batching
+	r.batchSize = opts.BatchSize
 	if opts.BatchMaxAge > 0 {
 		r.batchMaxAge = opts.BatchMaxAge
 	}
 	for _, v := range opts.Capabilities {
 		r.l.target.Capabilities = append(r.l.target.Capabilities, encoding.Symbol(v))
 	}
-	if opts.MaxCredit > 0 {
-		r.maxCredit = opts.MaxCredit
+	if opts.Credit > 0 {
+		r.l.linkCredit = uint32(opts.Credit)
+	} else if opts.Credit < 0 {
+		r.l.linkCredit = 0
```

Comment:
Won't this cause our 'rx' channel to be sized to zero?

```go
func (r *Receiver) attach(ctx context.Context) error {
    // ...
    r.l.rx = make(chan frames.FrameBody, r.l.linkCredit)
```

Reply:
You are right and is why I had removed the buffered channel in this PR (will add that back).

Reply:
Fixed.